### PR TITLE
Feature server side only login

### PR DESCRIPTION
**Pull request by flbulgarelli**
# Main components

# Helpers 

# Basic Usage 

## Rails

```ruby
# in initializers/omniauth.rb 
Rails.application.config.middleware.use OmniAuth::Builder do
   Mumukit::Login.configure_omniauth! self
end

# in config/routes.rb
Rails.application.routes.draw do
  Mumukit::Login.configure_login_routes! self
end

# in app/controllers/login_controller.rb
class LoginController < ApplicationController
  Mumukit::Login.configure_login_controller! self

  def failure
      # define your failure login handler
  end
end

# in app/controller/application_controller.rb
class ApplicationController < ActionController::Base
  Mumukit::Login.configure_controller! self
end
```

## Customization

You can override the following methods: 

* `login_methods`
* `destroy_session_user_uid!`, `save_session_user_uid!`, `current_user_uid`

### CODE REVIEW

**Comment by flbulgarelli** [comment]
@dggluz  please have a look at this

File: app/controllers/concerns/with_dynamic_errors.rb [hunk 1]
```diff
@@ -6,9 +6,10 @@ module WithDynamicErrors
     unless Rails.application.config.consider_all_requests_local
       rescue_from ActionController::RoutingError, with: :not_found
     end
+    #FIXME maybe we can user more Mumukit::Auth::Exceptions here
```
Comment (aguspina) [inline app/controllers/concerns/with_dynamic_errors.rb]: Definitely

**Comment by aguspina** [comment]
👍

**Comment by flbulgarelli** [comment]
TODO: review comments

**Comment by dggluz** [comment]
Looks good to me. :+1:

**Pull request merged**